src/calculateRentalCost.js

```diff
@@ -3,8 +3,28 @@
  *
  * @return {number}
  */
+const oneDayCost = 40;
+const longTerm = 7;
+const longTermDiscount = 50;
+const almostWeek = 3;
+const almostWeekDiscount = 20;
+
 function calculateRentalCost(days) {
-  // write code here
+  let totalAmount = oneDayCost * days;
+
+  if (days >= longTerm) {
+    totalAmount = totalAmount - longTermDiscount;
+
+    return totalAmount;
+  }
+
+  if (days >= almostWeek) {
+    totalAmount = totalAmount - almostWeekDiscount;
+
+    return totalAmount;
+  }
```

Comment:
The logic for applying discounts is flawed. Currently, if `days` is greater than or equal to `longTerm`, the function applies the `longTermDiscount` and returns immediately, preventing the `almostWeekDiscount` from being applied. You should adjust the logic to ensure that both discounts are considered, possibly by checking the `almostWeek` condition first or restructuring the conditions to allow both discounts to be applied when appropriate.